src/program.hpp

```diff
@@ -286,10 +319,18 @@ class cvk_entry_point {
 
     const std::vector<kernel_argument>& args() const { return m_args; }
 
+    const kernel_image_channel_metadata_map image_channel_metadata() const {
```

Comment:
Guessing you intended the return type to be a reference, didn't you?

Reply:
indeed